### ISSUE TITLE
Testing page "Sample app"(testing.zip) is wrong

### ISSUE THREAD

**Issue by TuanWorkMail** [angular/angular#29982]
# 📚 Docs or angular.io bug report

### Description

 On the Testing page the link "download example"(Sample app) will download a file named testing.zip which is nearly identical to the file specs.testing.zip from the link "download example"(Tests). 
It suppose to download a project without any test in it like in "Sample app" link but it download a complete project with all the tests instead.

### What's the affected URL?**
 https://angular.io/guide/testing

### Reproduction Steps**

Click on both "download example" link and go to src\app\demo of both, both contain demo.spec.ts and demo.testbed.spec.ts 
### Expected vs Actual Behavior**

testing.zip should not contain any *.spec.ts

### Anything else relevant?

If we go to "Sample App" and click the download button next to PROJECT, we will get a clean project without any *.spec.ts, which is the outcome that we want from the "download example" link

**Comment by m-henderson** [comment]
@brandonroberts this could go both ways. Is it the stack blitz sample app that is incorrect or is it the zip of the sample app that is incorrect?

**Comment by TuanWorkMail** [comment]
stack blitz is correct because it is only code.

download example (sample app) is incorrect because it contain all the tests.

testing.zip and specs.testing.zip are basically the same.

**Comment by brandonroberts** [comment]
I don't think the sample app is incorrect just because it includes the tests. It's actually easier to use the sample app zip file if you want to download and run it like a normal project and use `ng test`.

**Comment by TuanWorkMail** [comment]
I just want a vanilla project so I can follow the tutorial, but it already have all the answers.
Also if the 2 zip files are nearly identical, what is the point?

**Comment by brandonroberts** [comment]
@TuanWorkMail if you want to treat it like a tutorial I can see your point. Would you like to put in a PR to change it?

**Comment by TuanWorkMail** [comment]
when I click edit on the page, I don't see any link to the zip file, I only see this:

<live-example embedded-style>Sample app</live-example>

The zip file is hosted on angular.io, and I don't see any zip file in the repo so I think only you guy can change it.

**Comment by brandonroberts** [comment]
Zip files are generated based on a JSON configuration for each example. The testing one is here https://github.com/angular/angular/blob/master/aio/content/examples/testing/stackblitz.json

You can exclude the specs using a glob for the download by adding 

"!src/app/**/*.spec.ts"

to the in the JSON file. Once you open the PR the preview can be generated which includes the zip. You can also do this locally using the AIO readme https://github.com/angular/angular/tree/master/aio#developer-tasks

If you need more assistance, let me know.

**Comment by TuanWorkMail** [comment]
I add that line to the json file and it generate a correct zip file, but when I try to 'yarn start' it throw this error:

```
ERROR in src/app/model/testing/index.ts(1,15): error TS2307: Cannot find module './fake-hero.service'.
src/testing/jasmine-matchers.d.ts(3,67): error TS2694: Namespace 'jasmine' has no exported member 'CustomMatcher'.
src/testing/jasmine-matchers.ts(7,3): error TS2708: Cannot use namespace 'jasmine' as a value.
src/testing/jasmine-matchers.ts(12,32): error TS2694: Namespace 'jasmine' has no exported member 'CustomMatcher'.
src/testing/jasmine-matchers.ts(14,97): error TS2694: Namespace 'jasmine' has no exported member 'CustomMatcherResult'.
```

I don't remember if I was able to run this before, maybe I saw the project run on stackblitz?
The zip file is correct, I just not able to confirm if it working before I open a PR.

there are no fake-hero.service in the zip
tested on Windows 10 1903 and macOS Mojave, same errors
should I open another issue?

**Comment by frederikprijck** [comment]
Hey @brandonroberts, I just noticed that running `ng test` fails in both example apps.

This issue appears to be related to the same example app (even tho it addresses another problem with it). Before I create a PR to fix `ng test`, I wanted to double check if @TuanWorkMail is still working on these examples and is willing to add a fix for that as well.

If not, I'd be glad to pick it up and sent in a PR.

**Comment by TuanWorkMail** [comment]
No I'm not working on it, please do.

**Comment by brandonroberts** [comment]
Thanks @frederikprijck let us know if you need any assistance.

**Comment by frederikprijck** [comment]
@brandonroberts currently blocked by https://github.com/angular/angular/issues/30939#issuecomment-500564220 which is failing the sample apps tests.

**Comment by m-henderson** [comment]
@frederikprijck where did you get that avatar?

**Comment by frederikprijck** [comment]
@m-henderson, To be honost, I have no idea. I've been using it for a long time, I think I generated it on some site somewhere 10-ish years ago.

**Comment by rockument69** [comment]
The links in https://angular.io/guide/testing now go to StackBlitz. There is no download link (except from within StackBlitz). The Test project shows a few more files than the Sample App. @kapunahelewong : This is probably fixed and should be closed?

**Comment by atscott** [comment]
Closing as obsolete.
